### PR TITLE
Ignoring resources which have null ipconfig or frontendIPConfig

### PR DESCRIPTION
**Pull request by anaggar**

### CODE REVIEW

**Comment by KrishnaAdityaB** [comment]
#1604

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 1]
```diff
@@ -496,13 +496,16 @@ function Get-MachinesFqdnsForPublicIP
         #Map the ipc to the fqdn
         foreach($publicIp in $publicIPAddressResources)
         {
-            if(-not [string]::IsNullOrEmpty($publicIP.DnsSettings.Fqdn))
+            if(-not [string]::IsNullOrEmpty($publicIp.IpConfiguration.Id))
```
Comment (pavanadepu2) [inline Tasks/AzureFileCopy/Utility.ps1]: Good to have both the checks. Verify DnsSettings.Fqdn too.

File: Tasks/DeployAzureResourceGroup/Utility.ps1 [hunk 2]
```diff
@@ -506,13 +506,16 @@ function Get-MachinesFqdnsForLB
         #Map the public ip id to the fqdn
         foreach($publicIp in $publicIPAddressResources)
         {
-            if(-not [string]::IsNullOrEmpty($publicIP.DnsSettings.Fqdn))
+            if(-not [string]::IsNullOrEmpty($publicIp.IpConfiguration.Id))
```
Comment (kmkumaran) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: If all the publicIP in the loop turns out to be bad, you will exit with no entry in fqdnMap. Azure File copy needs at-least one valid IP or DNS to progress. Where is that check?

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 3]
```diff
@@ -576,13 +582,16 @@ function Get-MachinesFqdnsForLB
         #Get the NAT rule for a given ip id
         foreach($config in $frontEndIPConfigs)
         {
-            $fqdn = $fqdnMap[$config.PublicIpAddress.Id.ToLower()]
-            if(-not [string]::IsNullOrEmpty($fqdn))
+            if(-not [string]::IsNullOrEmpty($config.PublicIpAddress.Id))
             {
-                $fqdnMap.Remove($config.PublicIpAddress.Id.ToLower())
```
Comment (pavanadepu2) [inline Tasks/AzureFileCopy/Utility.ps1]: Isn't it applicable for AzureRG

**Comment by pavanadepu2** [comment]
Looks fine

File: Tasks/AzureFileCopy/Utility.ps1 (hunk 1 above)
Comment (anaggar) [inline Tasks/AzureFileCopy/Utility.ps1]: we are doing that also in next statement..

File: Tasks/DeployAzureResourceGroup/Utility.ps1 (hunk 2 above)
Comment (anaggar) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: we have that check already in Get-MachineNameFromId function

**Comment by pstricker** [comment]
Has this been released yet?

**Comment by ericsciple** [comment]
Based on when it was merged to master and the deployment schedule, I think it should be there.

**Pull request merged**